### ISSUE TITLE
Including the source in the project in XCode 7

### ISSUE THREAD

**Issue by vnl** [frankosterfeld/qtkeychain#74]
I am trying to include qtkeychain within my project. I am quite new to XCode and also Mac. Not sure if I am missing something or this is an issue which can be fixed. 
 I have included the framework too. Compiling it directly from github clone works well. It is when I use the mac related files within my project that causes the following problems. 

`Undefined symbols for architecture x86_64:
  "QKeychain::ReadPasswordJobPrivate::staticMetaObject", referenced from:
      QKeychain::ReadPasswordJobPrivate::tr(char const*, char const*, int) in keychain_mac.o
  "QKeychain::Job::finished(QKeychain::Job*)", referenced from:
      QKeychain::Job::emitFinished() in keychain.o
  "vtable for QKeychain::JobPrivate", referenced from:
      QKeychain::JobPrivate::JobPrivate(QString const&, QKeychain::Job*) in keychain.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for QKeychain::JobExecutor", referenced from:
      QKeychain::JobExecutor::JobExecutor() in keychain.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for QKeychain::ReadPasswordJob", referenced from:
      QKeychain::ReadPasswordJob::ReadPasswordJob(QString const&, QObject*) in keychain.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for QKeychain::WritePasswordJob", referenced from:
      QKeychain::WritePasswordJob::WritePasswordJob(QString const&, QObject*) in keychain.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for QKeychain::DeletePasswordJob", referenced from:
      QKeychain::DeletePasswordJob::DeletePasswordJob(QString const&, QObject*) in keychain.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for QKeychain::ReadPasswordJobPrivate", referenced from:
      QKeychain::ReadPasswordJobPrivate::ReadPasswordJobPrivate(QString const&, QKeychain::ReadPasswordJob*) in keychain.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for QKeychain::WritePasswordJobPrivate", referenced from:
      QKeychain::WritePasswordJobPrivate::WritePasswordJobPrivate(QString const&, QKeychain::WritePasswordJob*) in keychain.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for QKeychain::DeletePasswordJobPrivate", referenced from:
      QKeychain::DeletePasswordJobPrivate::DeletePasswordJobPrivate(QString const&, QKeychain::DeletePasswordJob*) in keychain.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
  "vtable for QKeychain::Job", referenced from:
      QKeychain::Job::Job(QKeychain::JobPrivate*, QObject*) in keychain.o
      QKeychain::Job::~Job() in keychain.o
  NOTE: a missing vtable usually means the first non-inline virtual member function has no definition.
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
 `
If I need to provide additional information please let me know. 
I am using : Xcode Version 7.3.1 (7D1014) and QMake version 3.0
Using Qt version 5.6.0

**Comment by frankosterfeld** [comment]
Are keychain.h and keychain_p.h in the HEADERS section of your .pro file? Looks to me like moc isn't run/the moc-generated code not compiled/linked.

**Comment by vnl** [comment]
Thank you for the quick reply. That was a good call, I did not have it in the pri file. regenerated the file and now most of the errors have gone. Still have a few though.
`Undefined symbols for architecture x86_64:
  "_SecCopyErrorMessageString", referenced from:
      strForStatus(int) in keychain_mac.o
  "_SecKeychainAddGenericPassword", referenced from:
      writeEntryImpl(QString const&, QString const&, QByteArray const&, QString*) in keychain_mac.o
  "_SecKeychainFindGenericPassword", referenced from:
      readPw(QByteArray*, QString const&, QString const&, OpaqueSecKeychainItemRef**) in keychain_mac.o
  "_SecKeychainItemDelete", referenced from:
      deleteEntryImpl(QString const&, QString const&, QString*) in keychain_mac.o
  "_SecKeychainItemFreeContent", referenced from:
      readPw(QByteArray*, QString const&, QString const&, OpaqueSecKeychainItemRef**) in keychain_mac.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
`
I have added all the 3 header files to moc. Am I still missing something?
